From 1e64e270e092f8efdad07c668fda7c1d1edf473e Mon Sep 17 00:00:00 2001
From: Haru Zheng <haruzheng@realtek.com>
Date: Wed, 10 Nov 2021 16:40:11 +0800
Subject: [PATCH 40/98] usb: rtk: add the ftdi-spi driver.

[REVIEWER] Haru Zheng
[Information]
Add realtek energy probe kit support.
Default build at module.
Example on Stark:
cat /dev/ftdi_spi_device_*

Change-Id: I5063b702a0aa75c65db3d0e03ef630905a0eca30
---
 drivers/usb/misc/Kconfig  | 11 +++++++++++
 drivers/usb/misc/Makefile |  3 +++
 2 files changed, 14 insertions(+)

diff --git a/drivers/usb/misc/Kconfig b/drivers/usb/misc/Kconfig
index 2a10d4e3dbb6..3cca5c58798b 100644
--- a/drivers/usb/misc/Kconfig
+++ b/drivers/usb/misc/Kconfig
@@ -285,6 +285,17 @@ config BRCM_USB_PINMAP
 	  signals, which are typically on dedicated pins on the chip,
 	  to any gpio.
 
+config USB_FTDI_SPI
+       tristate "Realtek Energy Probe Kit Support"
+       depends on USB
+       default m
+       help
+         Support Realtek energy probe kit, need usb and board.
+	 Can get Voltage and current data and infomation summary
+	 by /dev/ftdi_spi_device_*.
+	 Future are support SPI mode function.
+
+
 config USB_ONBOARD_HUB
 	tristate "Onboard USB hub support"
 	depends on OF || COMPILE_TEST
diff --git a/drivers/usb/misc/Makefile b/drivers/usb/misc/Makefile
index 2c5aec6f1b26..ccd1c924d88a 100644
--- a/drivers/usb/misc/Makefile
+++ b/drivers/usb/misc/Makefile
@@ -32,4 +32,7 @@ obj-$(CONFIG_USB_CHAOSKEY)		+= chaoskey.o
 obj-$(CONFIG_USB_SISUSBVGA)		+= sisusbvga/
 obj-$(CONFIG_USB_LINK_LAYER_TEST)	+= lvstest.o
 obj-$(CONFIG_BRCM_USB_PINMAP)		+= brcmstb-usb-pinmap.o
+
+obj-$(CONFIG_USB_FTDI_SPI)		+= ftdi-spi.o
+
 obj-$(CONFIG_USB_ONBOARD_HUB)		+= onboard_usb_hub.o
-- 
2.45.2

